R/stan_module.R

```diff
@@ -0,0 +1,126 @@
+#' StanAll class object
+#' @slot functions Character, the functions part of a stan model.
+#' @slot data Character, the data part of a stan model.
+#' @slot parameters Character, the parameters part of a stan model.
+#' @slot transformed_parameters Character, the transformed_parameters part of a stan model.
+#' @slot model Character, the model part of a stan model.
+#' @slot priors List, the priors part of a stan model.
+#' @slot generated_quantities Character, the generated_quantities part of a stan model.
+#' @slot includes Character
+#' @slot inits List with the initial values of the stan model.
+#' @exportClass StanModule
+StanModule <- setClass(
+    "StanModule",
+    representation(
+        functions = "character",
+        data = "character",
+        parameters = "character",
+        transformed_parameters = "character",
+        priors = "list",
+        generated_quantities = "character",
+        inits = "list"
+    )
+)
+
+
+
+#' read_stan_part returns stan code as a character.
+#' @param file Character, either the absolute path of a stan file, or the name of the stan file in the package directory or the stan code as a string.
+#' @export
+read_stan_part <- function(file) {
+    if (file.exists(file)) {
+        absolute_filename <- file
+        out <- readLines(absolute_filename)
+    } else if (file.exists(system.file("stanparts", file, package = "jmpost"))) {
+        absolute_filename <- system.file(
+            "stanparts",
+            file,
+            package = "jmpost"
+        )
```

Comment:
You end up calling the system.file function twice here with identical arguments, perhaps just call it once in advance and assign the value to a variable.